test/e2erp/fakerp_test.go

```diff
@@ -3,15 +3,55 @@
 package e2erp
 
 import (
+	"reflect"
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
 var _ = Describe("Resource provider e2e tests [Fake]", func() {
 	defer GinkgoRecover()
 
-	It("dummy fake test", func() {
-		var err error
+	It("should be possible to maintain a healthy cluster after rotating all credentials", func() {
+		By("Parsing the external manifest")
+		external, err := parseExternalConfig(*manifest)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(external).NotTo(BeNil())
+
+		By("Parsing the internal manifest containing config blob")
+		internal, err := parseInternalConfig(*configBlob)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(internal).NotTo(BeNil())
+
+		By("Deleting all non-ca cluster certificates and credentials from the config blob")
+		mutated := deleteCertificates(internal)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(mutated).NotTo(BeNil())
+
+		By("Running generate on the modified config blob")
+		err = generateInternalConfig(mutated, pluginConfig)
 		Expect(err).NotTo(HaveOccurred())
+
+		By("Persisting the config blob containing the new certificates and credentials")
+		err = saveConfig(mutated, *configBlob)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Calling update on the fake rp with the updated config blob")
+		updated, err := updateCluster(ctx, external, *configBlob, logger, pluginConfig)
```

Comment:
The problem with running the upgrade here is that you have created your own way of upgrading clusters `updateCluster` that is different from what we run in development. That's why I have been working in enabling us to use createorupdate for running upgrade tests in an e2e way and it's now possible with https://github.com/openshift/openshift-azure/pull/730 + https://github.com/openshift/release/pull/2161

Not saying that you should change what you have now but long-term it's going to be more robust to have to maintain a single way to do upgrades (`createorupdate.go`) vs littering that code inside the e2e suites.

Reply:
Instead of having to write bash scripts that would need to run before and after the upgrade, we could still write ginkgo tests that get run via `ginkgo`. Or maybe we could reuse the code between createorupdate.go and here?